src/Encode/AlphaEncodingManager.cpp

```diff
@@ -1,111 +1,238 @@
+    {"(AAB)", VRCommDataAlphaEncodingKey::FinJointThumb1},   // thumb joint 1
+    {"(AAC)", VRCommDataAlphaEncodingKey::FinJointThumb2},   // thumb joint 2
+    {"(BAA)", VRCommDataAlphaEncodingKey::FinJointIndex0},   // index joint 0
+    {"(BAB)", VRCommDataAlphaEncodingKey::FinJointIndex1},   // index joint 1
+    {"(BAC)", VRCommDataAlphaEncodingKey::FinJointIndex2},   // index joint 2
+    {"(BAD)", VRCommDataAlphaEncodingKey::FinJointIndex3},   // index joint 3
+    {"(CAA)", VRCommDataAlphaEncodingKey::FinJointMiddle0},  // middle joint 0
+    {"(CAB)", VRCommDataAlphaEncodingKey::FinJointMiddle1},  // middle joint 1
+    {"(CAC)", VRCommDataAlphaEncodingKey::FinJointMiddle2},  // middle joint 2
+    {"(CAD)", VRCommDataAlphaEncodingKey::FinJointMiddle3},  // middle joint 3
+    {"(DAA)", VRCommDataAlphaEncodingKey::FinJointRing0},    // ring joint 0
+    {"(DAB)", VRCommDataAlphaEncodingKey::FinJointRing1},    // ring joint 1
+    {"(DAC)", VRCommDataAlphaEncodingKey::FinJointRing2},    // ring joint 2
+    {"(DAD)", VRCommDataAlphaEncodingKey::FinJointRing3},    // ring joint 3
+    {"(EAA)", VRCommDataAlphaEncodingKey::FinJointPinky0},   // pinky joint 0
+    {"(EAB)", VRCommDataAlphaEncodingKey::FinJointPinky1},   // pinky joint 1
+    {"(EAC)", VRCommDataAlphaEncodingKey::FinJointPinky2},   // pinky joint 2
+    {"(EAD)", VRCommDataAlphaEncodingKey::FinJointPinky3},   // pinky joint 3
+
+    {"(AB)", VRCommDataAlphaEncodingKey::FinSplayThumb},   // whole thumb splay
+    {"(BB)", VRCommDataAlphaEncodingKey::FinSplayIndex},   // whole index splay
+    {"(CB)", VRCommDataAlphaEncodingKey::FinSplayMiddle},  // whole middle splay
+    {"(DB)", VRCommDataAlphaEncodingKey::FinSplayRing},    // whole ring splay
+    {"(EB)", VRCommDataAlphaEncodingKey::FinSplayPinky},   // whole pinky splay
+
+    {"F", VRCommDataAlphaEncodingKey::JoyX},      // joystick x component
+    {"G", VRCommDataAlphaEncodingKey::JoyY},      // joystick y component
+    {"H", VRCommDataAlphaEncodingKey::JoyBtn},    // joystick button
+    {"I", VRCommDataAlphaEncodingKey::BtnTrg},    // trigger button
+    {"J", VRCommDataAlphaEncodingKey::BtnA},      // A button
+    {"K", VRCommDataAlphaEncodingKey::BtnB},      // B button
+    {"L", VRCommDataAlphaEncodingKey::GesGrab},   // grab gesture (boolean)
+    {"M", VRCommDataAlphaEncodingKey::GesPinch},  // pinch gesture (boolean)
+    {"N", VRCommDataAlphaEncodingKey::BtnMenu},   // system button pressed (opens SteamVR menu)
+    {"O", VRCommDataAlphaEncodingKey::BtnCalib},  // calibration button
+    {"", VRCommDataAlphaEncodingKey::Null},       // Junk key
 };
 
-static std::string getArgumentSubstring(const std::string& str, const char del) {
-  const size_t start = str.find(del);
+static const std::map<VRCommDataAlphaEncodingKey, std::string> VRCommDataAlphaEncodingOutputKeyString{
+    {VRCommDataAlphaEncodingKey::FinThumb, "A"},   // thumb force feedback
```

Comment:
Might be a good idea to have inputs and outputs be two separate maps. (i.e. have force feedback keys be on a separate map)

Reply:
Not super familiar with maps though so I could be wrong.

Reply:
yeah we could, for now we only have outputs which are in the map of inputs, but it's simple to split it up when needed.

Reply:
Want to go ahead and split it up now?

Reply:
😭

Reply:
sure

Reply:
LMAO, I already added in an output map, I'm blind.